### PR TITLE
Fix anthropic for async and streaming and autowrap in tests

### PR DESCRIPTION
**Pull request by adam-baserun**

### CODE REVIEW

File: baserun/instrumentation/anthropic.py [hunk 1]
```diff
@@ -77,9 +77,29 @@ def set_response_attributes(span: _Span, response):
         span.set_attribute(SpanAttributes.ANTHROPIC_LOG_ID, response.log_id)
 
     @staticmethod
-    def generator_wrapper(original_generator: GeneratorType, span: _Span):
-        raise NotImplementedError
+    def generator_wrapper(original_generator: collections.abc.Iterator, span: _Span):
```
Comment (adam-baserun) [inline baserun/instrumentation/anthropic.py:80]: Just basic implementation in anthropic

File: baserun/instrumentation/base_instrumentor.py [hunk 2]
```diff
@@ -57,7 +57,11 @@ def _instrument(self, **kwargs):
                 f"{original_class.__name__}.{original_method.__name__}"
             ] = original_method
 
-            if inspect.iscoroutinefunction(original_method):
+            unwrapped_method = original_method
```
Comment (adam-baserun) [inline baserun/instrumentation/base_instrumentor.py:60]: We solved this in the previous version by unwrapping, but this check was failing for anthropic's async functions that were already wrapped

File: baserun/instrumentation/instrumented_wrapper.py [hunk 3]
```diff
@@ -87,7 +87,8 @@ async def instrumented_function(*args, **kwargs):
         finally:
             if auto_end_span:
                 span.end()
-                parent_span.end()
+                if parent_span.is_recording() and not Baserun.current_test_suite:
```
Comment (adam-baserun) [inline baserun/instrumentation/instrumented_wrapper.py:90]: I assume this should check is_recording like we do below?

File: baserun/instrumentation/instrumented_wrapper.py [hunk 4]
```diff
@@ -37,7 +37,7 @@ async def instrumented_function(*args, **kwargs):
         tracer = tracer_provider.get_tracer("baserun")
 
         parent_span: _Span = get_current_span()
-        if not parent_span.is_recording():
+        if not parent_span.is_recording() and not Baserun.current_test_suite:
```
Comment (adam-baserun) [inline baserun/instrumentation/instrumented_wrapper.py:40]: Fixes for auto wrap to not do so during a test run

File: baserun/instrumentation/instrumented_wrapper.py [hunk 5]
```diff
@@ -152,7 +153,7 @@ def instrumented_function(*args, **kwargs):
                     raise e
 
                 # If this is a streaming response, wrap it, so we can capture each chunk
-                if isinstance(response, GeneratorType):
+                if isinstance(response, Iterator):
```
Comment (adam-baserun) [inline baserun/instrumentation/instrumented_wrapper.py:156]: Anthropic's response is not a generator, updated this and above to check if it is an Iterator / AsyncIterator

File: baserun/instrumentation/instrumented_wrapper.py (hunk 3 above)
Comment (erik-megarad) [reply baserun/instrumentation/instrumented_wrapper.py:90]: Yeah good catch

**Review by erik-megarad — APPROVED** [review]

**Pull request merged**